### PR TITLE
LLF cloud top height

### PR DESCRIPTION
**Pull request by mpartio**
Review of low level forecast cloud top height.

See: https://jira.fmi.fi/browse/STU-19510

### CODE REVIEW

**Review by tackandr — COMMENTED** [review]
This follows the makro version correctly. At least I couldn't find any differences.
I would have more feedback on the makro in general but don't know if there is any interest to change that. For example I think it would be cleaner to do the algorithm in pressure based vertical coordinates and convert the results from pressure to FL in the end instead of using height based coordinate. I think height based also introduces some error as conversion between FL and hft is not a constant but can't quantify how big that error would be. It might be negligible.
Another question would be why below FL050 height is given over ground and not over sea level because I think aircraft instrumentation is in reference to sea level so I would expect aviation weather to be reported accordingly.
But this is feedback to Simo rather than to you.

**Comment by KorpinenA** [comment]
I have checked both macro version and now lua version and I didn't would any issues or differences with a calculation logic. So I think it is ok to merge this pull request.

File: himan-scripts/llf.lua [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  local thresholddata = {}
+
+  for i = 1, #qfedata do
+     zerodata[i] = 0
+     maxHdata[i] = maxH
+     thresholddata[i] = threshold
+
+     -- surface light level in meters (can be negative)
+     -- use pressure in hPa
+     sfcFLmdata[i] = (3.7314 - (0.01 * qfedata[i]) ^ (1.0 / 5.2559)) / 0.0000256 * 0.3048
```
[the remark this answers is not in the document]
Comment (mpartio) [reply himan-scripts/llf.lua:74]: I tried that function but it gave very different results than the formula in the macro.

[the remark this answers is not in the document]
Comment (tackandr) [reply himan-scripts/llf.lua:74]: I verified the Himan function and it gives correct flight levels. Input is Pa though. I should add that documentation to the function. I can see now the comment above that you want flight level in meters. So metutil function(QFE in Pa) * 100 * 0.3048 should give that.

[the remark this answers is not in the document]
Comment (mpartio) [reply himan-scripts/llf.lua:74]: I tried exactly that, but there are (big) differences. For one, Simo's formula gives negative Flight Levels, Himan version does not.

I'm not claiming which one is the "correct one" (probably the one in Himan), but I'm a bit afraid of changing the formula as the results are so different.

**Pull request merged**